packages/fs-detectors/src/detect-builders.ts

```diff
@@ -305,6 +305,27 @@ export async function detectBuilders(
     };
   }
 
+  // Exclude the middleware builder for Next.js apps since @vercel/next
+  // will build middlewares.
+  //
+  // While maybeGetApiBuilder() excludes the middleware builder, however,
+  // we need to check if it's a Next.js app here again for the case where
+  // `projectSettings.framework == null`.
+  if (
+    framework === null &&
+    detectFrontBuilder(pkg, files, usedFunctions, fallbackEntrypoint, options)
+      .use === '@vercel/next' &&
```

Comment:
```suggestion
    frontendBuilder?.use === '@vercel/next' &&
```

This can be simplified to use the result from detectFrontBuilder above

Reply:
Fixed. I used `detectFrontBuilder` to reuse the `createdAt < Date.parse('2020-03-01')` condition. I copied that into this if statement instead.

Reply:
I don’t think you need to check the date here since it’s already checked when frontendBuilder is assigned